sql/api/src/main/scala/org/apache/spark/sql/catalyst/encoders/AgnosticEncoder.scala

```diff
@@ -286,5 +286,6 @@ object AgnosticEncoders {
     override def dataType: DataType = transformed.dataType
     override def schema: StructType = transformed.schema
     override def isStruct: Boolean = transformed.isStruct
+    override def nullable: Boolean = transformed.nullable
```

Comment:
yikes...

Reply:
Given this is only for Spark 4 I'd be happier adding nullable as a param, this would allow signalling to the builders if the codec can handle nulls or not for the general case.  Let me know if that's an option and I'll do a full suite of testing again.